fecgov/fec-cms
# 4176 passive listeners

## Summary

- Resolves #4176 

Converting listeners to `passive` (when/where possible) to improve page performance. As far as our performance goes, passive listeners only matter when looking at touchstart and scroll events.

### Required reviewers

UX and/or front-end devs

## Impacted areas of the application

Because passive listeners are a breaking change for [browsers that don't support them](https://caniuse.com/passive-event-listener), implementing them is less straight-forward that one would expect.

We're getting flagged for non-passive listeners coming from our vendor file, recaptcha, and contributions-by-state.
- vendor
  - Updated leaflet, hoping that will alleviate the warning
  - An upcoming [major version of jQuery](https://github.com/jquery/jquery/issues/2871) will support passive listeners but it's a big change and unwieldy with the current version. jQuery 4.0 has no expected release date; it's [currently estimated at 73% complete](https://github.com/jquery/jquery/milestone/7)
- Recaptcha is a work in progress—looking into options
- Where Contributions Come From has been updated to use passive listeners
- The presidential finance map has been updated to use passive listeners

## Screenshots

Should be no visible changes

## Related PRs

## How to test

This should not change anything in most modern browsers, apart from improve performance while scrolling, particularly for touch devices.

Because they could break older browsers, we need to test in Internet Explorer, Chrome ≤ 50, Firefox ≤ 48, Safari < 10 (OS X and iOS), Edge ≤ 15

- pull the branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Make sure the JavaScript file for WCCF isn't included in the Lighthouse complaint about passive listeners
- Make sure the [presidential finance map](http://127.0.0.1:8000/data/candidates/president/presidential-map/) doesn't have its own entry in the Lighthouse passive listeners alert. **Also make sure the 'export raising' window scroll functionality works as expected**


## System architecture updates

None
